pkg/sharings/sharings.go

```diff
@@ -210,30 +210,53 @@ func ShareDoc(instance *instance.Instance, sharing *Sharing, recStatus *Recipien
 
 		// Dynamic sharing
 		if rule.Selector != "" {
-			// Create index based on selector to retrieve documents to share
-			indexName := "by-" + rule.Selector
-			index := mango.IndexOnFields(docType, indexName, []string{rule.Selector})
-			err := couchdb.DefineIndex(instance, index)
-			if err != nil {
-				return err
-			}
 
-			var docs []couchdb.JSONDoc
+			// Particular case for referenced_by: use the existing view
+			if rule.Selector == "referenced_by" {
+				// The permission tilte is used to know the referenced doctype
+				refType := rule.Title
```

Comment:
Mmmh, we could change the Values format for referenced_by rules. 
The rule would become VERB:io.cozy.files:io.cozy.album/45878584848485:referenced_by.

This will need handling both in sharing and vfs.File.Valid